### ISSUE TITLE
Facing issue while using autotune

### ISSUE THREAD

**Issue by nishantnnb** [kahst/BirdNET-Analyzer#203]
Hello,

I am trying to train birds sounds for aprox 200 classes **with autotune**. I am novice to ML & Python coding and got very basic idea of hyper-parameters.
Facing issue while training large data set. Issue details are listed below.

Installed keras-tuner.

Used following settings for training:
--model_format both 
--cache_mode save 
--cache_file _file_path_/x.npz 
--autotune

Following settings not entered explicitly (hence default values should be taken). 
--upsampling_ration, --upsampling_method,  --autotune_trials & --autotune_executions_per_trial 

Initially took trial on small data set for data of 10 birds.  **It was successful.** 
cache file got created.
various combinations of hyper-parameters for autotune used by the program for training.
model got created - tflite as well as raven

Hence started taking trial for full dataset of aprox 200 bird with same settings for training.
cache file got created.
following are the hyper-parameters got chosen for each trial. It stopped after 5th trial (without creating the model).
![image](https://github.com/kahst/BirdNET-Analyzer/assets/72615789/738deb9c-1c0a-4d0f-8a61-cb680ac55d5e)

Following are the error messages:
![image](https://github.com/kahst/BirdNET-Analyzer/assets/72615789/9f0694b3-68e7-4e72-a720-f05f44f69f45)

Is it something related to upsampling_ratio (as the trials chart & the error suggests) ? If yes, then why it worked OK for small dataset (10 birds)?

Can someone kindly help with this?

Thanks.
Nishant.

**Comment by max-mauermann** [comment]
Hi,
I looked into the issue and from what i can tell this occurs when a folder in the training data has no files in it.
Could you please verify that there are samples for each class in the training data?

**Comment by nishantnnb** [comment]
Yes indeed, found an empty folder for a class. 
Started training again now. Will revert back soon.
Thank you !

**Comment by nishantnnb** [comment]
Update: Training with autotune was successful after removing the empty class.
-- Cache file got saved.
-- Models got created (tflite as well as Raven).

But now when I am using this model (V001-**04**.tflite created **with autotune**) to analyze with Windows GUI, it gives me this error:
![image](https://github.com/kahst/BirdNET-Analyzer/assets/72615789/e4ef925f-2e16-47c9-9ea4-d9393a1eb4bc)

The file path shown in the error message (...local\temp...) is not the one where the sound file was stored. 

If I use my earlier custom model (V001-**03**.tflite) which was created **without Autotune**, no such error is given and results are shown properly. 
![image](https://github.com/kahst/BirdNET-Analyzer/assets/72615789/57f70ad9-010a-431f-9775-19c678219b7c)

Sorry for bothering too much. If you could kindly help with this.

Regards,
Nishant.

**Comment by nishantnnb** [comment]
Forgot to add this note: After completion of training with autotune, I had received such warning messages:
![image](https://github.com/kahst/BirdNET-Analyzer/assets/72615789/e9b6e085-fe1d-4f0b-b235-32697cb010b0)

And final hyperparameters were these:
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/nisha/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/nisha/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

Hyperparameter | Value
-- | --
Hidden   units | 512
Dropout | 0.25
Batchsize | 32
Learning rate | 0.0005
Crop mode | center
Crop overlap | 0
Upsamling mode | repeat
Upsamling   ratio | 0
use mixup | FALSE
use label   smoothing | FALSE



</body>

</html>

**Comment by nishantnnb** [comment]
Update: If I use the gui from command line with same custom model, no error given and prediction results are shown properly.

**Comment by nishantnnb** [comment]
Today reinstalled the Zip file for Windows GUI. Created new custom model using Windows GUI. Again same error while using the custom model for classification - with Single File option.  Then I used Multiple Files option. Txt files of the results generated successfully. 
Hence the issue might be with > Windows GUI > Single file tab.

Note: Single file classification works well when I open the GUI from command prompt (command: python gui.py)
